Core/include/Acts/Seeding/SeedFinder.ipp

```diff
@@ -771,7 +773,7 @@ SeedFinder<external_spacepoint_t, grid_t, platform_t>::filterCandidates(
       // convert p(T) to p scaling by sin^2(theta) AND scale by 1/sin^4(theta)
       // from rad to deltaCotTheta
       float p2scatterSigma = iHelixDiameter2 * sigmaSquaredPtDependent;
-      if (!std::isinf(m_config.maxPtScattering)) {
```

Comment:
Could we just check this in the `SeedFinder` constructor? There's a number of `isnan` checks there already?

Maybe `inf` is used as a sentinel value to switch this cut off, but we could maybe also use an `std::optional` to signal that the cut is unset.

Reply:
did you try using `std::isfinite`?

Reply:
I did not try, but in principle `std::isfinite` will check against `NaN` and not infinity, so I would naively say that might be the same effect.

Reply:
I cannot reproduce this in a micro benchmark https://quick-bench.com/q/-bf3LqAZSlpdQ56i958SVbKBCsI

I would also think that the branch predictor would get this always right since the config never changes after construction. Of course the CPU would still have to evaluate if the branch was correctly taken but compared to what else we do in the seeding I have a hard time to imagine why this would have an impact.

Potentially the profiler accumulated different costs on this line or just "randomly" associated it there.

What is the difference in runtime you measured with this change?

Reply:
Probably I might be just fooling myself and perhaps the profiler accumulated different costs as you suggests.  
The expected improvement in runtime is negligible and probably within noise. I just saw it popping up in the vtune (see screenshot) with variable percentage (0.1-0.3%) depending on config and thought might as well change it with a simpler check. 

I agree that I am surprised too to see it popping up: being a fixed value the processor shouldn't do mispredictions. 
Agree that it's super minor compared to other things in this code. 

<img width="781" alt="Screenshot 2025-02-18 at 21 24 21" src="https://github.com/user-attachments/assets/cfc9fdcb-bfbe-42f0-9a1f-536acdf95622" />